File: src/main/kotlin/com/lcaohoanq/sp/init/FirebaseInitializer.java
package com.lcaohoanq.sp.init;

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import java.io.ByteArrayInputStream;
import java.io.InputStream;
import javax.annotation.PostConstruct;
import org.springframework.stereotype.Component;

@Component
public class FirebaseInitializer {

    private static final String CONFIG = """
        {
        }
        """;

    @PostConstruct
    public void init() {
        try {
            InputStream serviceAccount = new ByteArrayInputStream(CONFIG.getBytes());

            FirebaseOptions options = FirebaseOptions.builder()
                .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                .build();

            if (FirebaseApp.getApps().isEmpty()) {
                FirebaseApp.initializeApp(options);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/com/lcaohoanq/sp/init/FirebaseInitializer.kt
import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import jakarta.annotation.PostConstruct
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class FirebaseInitializer {

    private val log = LoggerFactory.getLogger(FirebaseInitializer::class.java)

    @PostConstruct
    fun init() {
        try {
            val serviceAccount = javaClass.classLoader.getResourceAsStream("firebase-service-account.json")
                ?: throw IllegalStateException("firebase-service-account.json not found in resources")

            val options = FirebaseOptions.builder()
                .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                .build()

            if (FirebaseApp.getApps().isEmpty()) {
                FirebaseApp.initializeApp(options)
                log.info("Firebase has been initialized successfully.")
            } else {
                log.info("Firebase already initialized.")
            }

        } catch (e: Exception) {
            log.error("Failed to initialize Firebase", e)
        }
    }
}
